Source/Core/Core/Movie.cpp

```diff
@@ -597,6 +597,8 @@ static std::string Analog1DToString(u32 v, const std::string& prefix, u32 range
 // NOTE: CPU Thread
 static void SetInputDisplayString(ControllerState padState, int controllerID)
 {
+  if (controllerID >= 4)
```

Comment:
All of these boundary checks should either go in a named local function doing a boolean check or be checked against a named constant (probably the former). To anyone new to the codebase or unfamiliar with this area, it's likely not immediately obvious why the index would be greater than four.

Reply:
I've nuked these bounds-checks as they were mainly just a hack to deal with `s_InputDisplay` not being large enough initially and me not bothering with fixing that yet.  In the proper version I'm now using `SerialInterface::MAX_SI_CHANNELS` or `MAX_BBMOTES` as appropriate for shifts and loops which should be more descriptive.

(For some of the locations (the gamecube ones), it _wouldn't_ have been more than 4, but I didn't want to figure out which ones because that wasn't my immediate concern.)